app/helpers/news_helper.rb

```diff
@@ -7,11 +7,20 @@ def news_article_class
     'content-listing'
   end
 
-  def news_articles(section: nil)
-    if section.present?
-      section.news_articles.published.by_release_date.by_published_at
-    else
-      NewsArticle.published.by_release_date.by_published_at
-    end
+
+  # Loads news articles and preloads all associated relationships to be able to
+  # render the news page without any further database roundtrips.
+  #
+  # To load articles for a Section, use `preload_news_articles(section.news_articles)`
+  def preload_news_articles(query_root = NewsArticle)
+    query_root
+      .preload([
+        {:section  => {:home_node => :parent}},
+        {:sections => {:home_node => :parent}}
+      ])
+      .published
+      .by_release_date
+      .by_published_at
+      .by_name
```

Comment:
There was an existing spec that tests for order by dates then name. The previous implementation did not sort by name and it was accidentally passing. So I made it sort by name.